src/modules/hip/kernel/snow.cpp

```diff
@@ -102,58 +130,65 @@ extern "C" __global__ void snow_pln( unsigned char *output,
                 if (id_x + i + 5 <= width && id_y + j < height)
```

Comment:
this looks like a bug. I think id_x+i <= width is good. Please double check

Reply:
We'll do the kernel level optimizations subsequent PRs